core/src/main/java/org/apache/accumulo/core/client/impl/ConnectorImpl.java

```diff
@@ -193,4 +221,151 @@ public synchronized ReplicationOperations replicationOperations() {
+    public ConnectionOptions withBatchWriterConfig(BatchWriterConfig batchWriterConfig) {
+      info.setProperty(ClientProperty.BATCH_WRITER_MAX_MEMORY_BYTES, batchWriterConfig.getMaxMemory());
+      info.setProperty(ClientProperty.BATCH_WRITER_MAX_LATENCY_SEC, batchWriterConfig.getMaxLatency(TimeUnit.SECONDS));
+      info.setProperty(ClientProperty.BATCH_WRITER_MAX_TIMEOUT_SEC, batchWriterConfig.getTimeout(TimeUnit.SECONDS));
+      info.setProperty(ClientProperty.BATCH_WRITER_MAX_WRITE_THREADS, batchWriterConfig.getMaxWriteThreads());
+      info.setProperty(ClientProperty.BATCH_WRITER_DURABILITY, batchWriterConfig.getDurability().toString());
+      return this;
+    }
+
+    @Override
+    public SaslOptions withPrimary(String kerberosServerPrimary) {
+      info.setProperty(ClientProperty.SASL_KERBEROS_SERVER_PRIMARY, kerberosServerPrimary);
+      return this;
+    }
+
+    @Override
+    public SaslOptions withQop(String qualityOfProtection) {
+      info.setProperty(ClientProperty.SASL_QOP, qualityOfProtection);
+      return this;
+    }
+
+    @Override
+    public ConnectorFactory usingProperties(String configFile) {
+      Properties properties = new Properties();
+      try (InputStream is = new FileInputStream(configFile)) {
+        properties.load(is);
+      } catch (IOException e) {
+        throw new IllegalArgumentException(e);
+      }
+      return usingProperties(properties);
+    }
+
+    @Override
+    public ConnectorFactory usingProperties(Properties properties) {
+      info = new ConnectionInfoImpl(properties);
+      return this;
+    }
+
+    @Override
+    public ConnectionOptions usingBasicCredentials(String username, CharSequence password) {
+      info.setProperty(ClientProperty.AUTH_TYPE, "basic");
+      info.setProperty(ClientProperty.AUTH_BASIC_USERNAME, username);
+      info.setProperty(ClientProperty.AUTH_BASIC_PASSWORD, password.toString());
+      return this;
+    }
+
+    @Override
+    public ConnectionOptions usingKerberosCredentials(String principal, String keyTabFile) {
+      info.setProperty(ClientProperty.AUTH_TYPE, "kerberos");
+      info.setProperty(ClientProperty.AUTH_KERBEROS_PRINCIPAL, principal);
+      info.setProperty(ClientProperty.AUTH_KERBEROS_KEYTAB_PATH, keyTabFile);
+      return this;
+    }
+
+    @Override
+    public ConnectionOptions usingCredentials(String principal, AuthenticationToken token) {
+      if (token instanceof PasswordToken) {
+        return usingBasicCredentials(principal, new String(((PasswordToken) token).getPassword()));
+      } else if (token instanceof KerberosToken) {
+        return usingKerberosCredentials(principal, ((KerberosToken) token).getKeytab().getAbsolutePath());
+      } else {
+        throw new IllegalArgumentException("Unknown authentication token type");
+      }
+    }
+
+    @Override
+    public ConnectorFactory usingConnectionInfo(ConnectionInfo connectionInfo) {
+      this.info = (ConnectionInfoImpl) connectionInfo;
```

Comment:
Setting the type to ConnectionInfo should eliminate this cast.

Reply:
With 705d604759f9495cb, I no longer cast